File: C++/0067. Red and Green and White.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.
#include<vector>
#include<cmath>
int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    int n;
    while(cin>>n){
        string res(n+1,0);
        string temp(1,1);
        
        for(int i=1;i<=n;++i){
            int m = temp.size();
            for(int j=m-1;j>=0;j--){
                res[j+n+1-m] += temp[j]*3;
                res[j-1+n+1-m] += res[j+n+1-m]/10;
                res[j+n+1-m] = res[j+n+1-m]%10;
            }
            temp = res;
            res = string(n+1,0);          
        }
        res = temp;
        int flag = 1;
        for(int i=res.size()-1;i>=0;--i){
            res[i] += flag;
            flag = res[i]/10;
            res[i] %= 10;
            if(flag==0) break;
        }
        flag = 0;
        for(int i=0;i<res.size();++i){
            res[i] += 10*flag;
            flag = res[i]%2;
            res[i] /= 2;
        }
        
        for(auto&c:res) c+='0';
        int i;
        for(i=0;i<res.size();++i){
            if(res[i]!='0'){
                break;
            }
        }
        cout<<res.substr(i)<<endl;
    }
    return 0;
}

File: Python3/0067. Red and Green and White.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
#             n = int(input())
#             fac = [1]*((n+1)//2+1)
#             for i in range(1,(n+1)//2+1):
#                 fac[i] = fac[i-1]*(n-2*i+2)*(n-2*i+1)//(2*i-1)//(2*i)
#             res = 0
#             for i in range(0,n+1,2):
#                 res += fac[i//2]*pow(2,(n-i))
#             print(res)
            
            n = int(input())
            print((pow(3,n)+1)//2)
        
        except EOFError:
            break
    
if __name__ == "__main__":
    func()
